### ISSUE TITLE
Gitlab driver expects visibility from API, but it does not exist if repository is public and client is not authendicated

### ISSUE THREAD

**Issue by mikk150** [composer/composer#6642]
My `composer.json`:

```json
{
    "minimum-stability": "dev",
    "require": {
        "yiisoft/yii2-queue": "^2.2.0",
    },
    "repositories": [
        {
            "type": "vcs",
            "url": "https://gitlab.com/yii2-queue/yii2-queue"
        },
    ]
}

```

Output of `composer diagnose`:

```
# composer diagnose
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
```

When I run this command:

```
composer update
```

I get the following output:

```
                               
  [ErrorException]             
  Undefined index: visibility  
                               

```

And I expected this to happen:

should update my dependencies

gitlab does not tell if the project is public of private, if user is not authendicated... prolly because it seeing it while not being authendicated means it HAS to be public

**Comment by mlncn** [comment]
Very very excited to have this fixed!  This seems to be the last step in finally being able to have GitLab projects managed by composer.

**Comment by Exadra37** [comment]
Same error here when running `composer install` from docker.

#### Composer File

```json
{
    "name": "exadra37-docker-compose/laravel5-demo",
    "description": "A Laravel 5 demo to show how to use exadra37-docker-compose/laravel5.",
    "keywords": [
        "Exadra37",
        "laravel 5",
        "docker",
        "docker compose",
        "laravel 5 docker compose demo"
    ],
    "homepage": "https://gitlab.com/exadra37-docker-compose/laravel5-demo",
    "license": "GPL-3.0",
    "authors": [
        {
            "name": "Paulo Silva",
            "email": "exadra37ingmailpointcom",
            "homepage": "https://exadra37.com",
            "role": "Software Developer"
        }
    ],
    "repositories": [
        {
            "type": "vcs",
            "url": "git@gitlab.com:exadra37-docker-compose/laravel5.git"
        }
    ],
    "require": {
        "php": ">=7.0",
        "exadra37-docker-compose/laravel5": "dev-optimize-nginx"
    },
    "require-dev": {
        "bmitch/codor": "^1.0",
        "laravel/laravel": "5.4.*",
        "phploc/phploc": "^3.0",
        "phpmd/phpmd": "^2.6",
        "phpspec/phpspec": "^3.2",
        "phpunit/phpunit": "^6.0",
        "sebastian/phpcpd": "^3.0",
        "sensiolabs/security-checker": "^4.0",
        "slevomat/coding-standard": "^2.0",
        "squizlabs/php_codesniffer": "^2.8",
        "symplify/easy-coding-standard": "^1.4"
    },
    "autoload": {
        "psr-4": {
            "Exadra37DockerCompose\\Laravel5Demo\\": "src/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "spec\\Exadra37DockerCompose\\Laravel5Demo\\": "tests/PhpSpec/spec",
            "tests\\Tools\\Exadra37DockerCompose\\Laravel5Demo\\": "tests/Tools"
        }
    }
}
```
#### Composer Error

![screenshot from 2017-09-02 10-38-35](https://user-images.githubusercontent.com/1745456/29994596-05cb85ce-8fcb-11e7-9073-c54d6625824b.png)

**Comment by Seldaek** [comment]
Fixed by #6643
